AK/HashTable.h

```diff
@@ -222,14 +206,15 @@ class HashTable {
 
     void ensure_capacity(size_t capacity)
     {
-        VERIFY(capacity >= size());
-        rehash(capacity * 2);
+        MUST(try_ensure_capacity(capacity));
     }
 
     ErrorOr<void> try_ensure_capacity(size_t capacity)
     {
         VERIFY(capacity >= size());
-        return try_rehash(capacity * 2);
+        if (capacity <= m_capacity)
+            return {};
+        return try_rehash(capacity * 100 / grow_at_load_factor_percent);
```

Comment:
The `grow_at_load_factor_percent` here is confusing to me, since it is not used as a percent.

This will essentially make it `capacity * 1.2`, is that the intention?
Maybe a comment / making a different constant with the same value would make this more clear.

Reply:
The goal here is to ensure enough capacity that you can set all values without the `HashTable` needing to grow/allocate again. That's why the `grow_at_load_factor_percent` is inverted.

I'll add a comment, but I also need to fix the logic since we might still need to allocate at the very last value since the check is whether the load factor of the table is larger than _or equal to_ `grow_at_load_factor_percent`.

Reply:
That makes sense, yeah a comment sounds like enough information for any future hashmap enthusiast

Reply:
I've fixed the calculation, and added a comment explaining why we're doing it at all.